Library/Homebrew/missing_formula.rb

```diff
@@ -198,6 +200,7 @@ def deleted_reason(name, silent: false)
 
       def cask_reason(name, silent: false, show_info: false)
         return if silent
+        return unless OS.mac?
```

Comment:
Ideally this would be a no-op method and have this method implementation moved to `Library/Homebrew/extend/os/mac/missing_formula.rb` instead.

Reply:
Sorry, I'm not sure I understand what you mean?

Reply:
Check out the other files in `Library/Homebrew/extend/os/mac`. We generally prefer taking that approach of extending the initial class to adding `OS.mac?` checks around the codebase.

Reply:
Thanks. I've done this in 5ecf7f5 which I'll squash at the end if everything's OK.